src/Core/Resolvers/Factories/IQueryManagerFactory.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.DataApiBuilder.Config.ObjectModel;
+
+namespace Azure.DataApiBuilder.Core.Resolvers.Factories
+{
+    /// <summary>
+    /// QueryManager Factory.
+    /// Used to get IQueryBuilder,IQueryExecutor and DbExceptionParser based on database type.
+    /// </summary>
+    public interface IQueryManagerFactory
```

Comment:
Nit: Should we rename the interface to something like `IAbstractQueryManagerFactory`? Going by definition, this follows abstract factory pattern.

Reply:
I think keeping it as IQueryManagerFactory should be fine to be in line with other naming patterns through the repository.

Reply:
But are any of those abstract factories though? Bringing this up so that we follow general good dev practices.

Reply:
That is fair. renamed this.